src/Analysis/Engine/Impl/AnalysisUnit.cs

```diff
@@ -342,6 +344,22 @@ internal AnalysisUnit(ScopeStatement ast, IScope scope)
         internal virtual ILocationResolver AlternateResolver => null;
 
         ILocationResolver ILocationResolver.GetAlternateResolver() => AlternateResolver;
+
+        readonly Lazy<AnalysisUnit> _externalAnnotationAnalysisUnit;
+        protected internal virtual AnalysisUnit GetExternalAnnotationAnalysisUnit() {
+            if (Ast != Tree)
```

Comment:
Since `_externalAnnotationAnalysisUnit` is only created when `ast == tree` it looks like this can be simplified to `=> _externalAnnotationAnalysisUnit != null ? _externalAnnotationAnalysisUnit.Value : null;`

Reply:
accepted this suggestion